### PR TITLE
Update dependencies and owner

### PR DESCRIPTION
**Pull request by nprokopic**
- Update owner to Team Phoenix
- Update base container image to quay.io/giantswarm/alpine:3.16.2-giantswarm.
- Update go to 1.18.
- Update github.com/giantswarm/k8sclient to v7.0.1.
- Update github.com/hashicorp/vault/api to v1.7.2.
- Update github.com/prometheus/client_golang to v1.13.0.
- Update github.com/spf13/afero to v1.9.2.
- Update k8s.io/api to v0.23.10.
- Update k8s.io/apimachinery to v0.23.10.
- Update k8s.io/client-go to v0.23.10.

### CODE REVIEW

File: helm/cert-exporter/Chart.yaml [hunk 1]
```diff
@@ -7,4 +7,5 @@ home: https://github.com/giantswarm/cert-exporter
 version: 2.1.0
 appVersion: 2.0.0
 annotations:
+  application.giantswarm.io/team: "hydra"
```
Comment (fiunchinho) [inline helm/cert-exporter/Chart.yaml:10]: @alex-dabija should we own `cert-exporter`?

Comment (alex-dabija) [reply helm/cert-exporter/Chart.yaml:10]: No we shouldn't. According to the latest updates, which might not have been propagated yet to the repos and the intranet ownership page, it's actuall @giantswarm/team-phoenix which owns cert-exporter. See [doc](https://docs.google.com/spreadsheets/d/1GNXAbYo_HyjjRusd2KTGttuORBkX9iMdgnNp-xTQ9sw/edit#gid=0).

Comment (T-Kukawka) [reply helm/cert-exporter/Chart.yaml:10]: we appreciate and support any contributions whatsoever or claiming ownership:trollface:

File: .gitignore [hunk 2]
```diff
@@ -4,6 +4,7 @@
 !vendor/**
 
 .vscode
+/.idea
```
[the remark this answers is not in the document]
Comment (nprokopic) [reply .gitignore:7]: c'mon, `.vscode` was already there :grimacing: 

I can add this in a separate PR if you insist :smile:

File: helm/cert-exporter/Chart.yaml (hunk 1 above)
Comment (nprokopic) [reply helm/cert-exporter/Chart.yaml]: Well, it sparked the discussion :smile: Fixed now :)

**Review by zewolfe — APPROVED** [review]
LGTM

**Review by fiunchinho — APPROVED** [review]
Nice job

**Pull request merged**